### ISSUE TITLE
Minor Issues with Weston Package

### ISSUE THREAD

**Issue by superloach** [chromebrew/chromebrew#3284]
After installing, `weston` has issues with permissions on `XDG_RUNTIME_DIR`. On my system they're 755, instead of 700.
```
[14:25:29.238] warning: XDG_RUNTIME_DIR "/var/run/chrome" is not configured
correctly.  Unix access mode must be 0700 (current mode is 755),
and must be owned by the user (current owner is UID 1000).
```
It is also missing the dependency `libwebp`.
```
[14:26:09.660] Loading module '/usr/local/lib64/libweston-4/wayland-backend.so'
[14:26:09.662] Failed to load module: libwebp.so.7: cannot open shared object file: No such file or directory
[14:26:09.662] fatal: failed to create compositor backend
```
In addition, there's a warning at the end of building that states `remember to run 'libtool --finish /usr/local/lib64/weston'`. That should probably be added as well.

-----
I'll probably do a PR for this myself, just wanted to open an issue to track it in case I do something wrong 😛

**Comment by JL2210** [comment]
The `remember to run 'libtool --finish /usr/local/lib64/weston'` part is handled automatically. Nothing needs to be done about it. It's mainly for the developers of the package.

**Comment by JL2210** [comment]
What channel are you on?

**Comment by superloach** [comment]
Ah, alright.
I'm on Stable 74.0.3729.125, why do you ask?

**Comment by JL2210** [comment]
Just checking. The permissions seem similar for me:
```
chronos@localhost ~/Downloads $ ls -la /var/run/chrome/
total 0
drwxr-xr-x.  2 chronos chronos  80 May 20 08:53 .
drwxr-xr-x. 39 root    root    980 May 19 14:06 ..
srw-rw----.  1 chronos wayland   0 May 19 14:05 wayland-0
-rw-r-----.  1 chronos chronos   0 May 19 14:05 wayland-0.lock
```

Why not update to Weston 5?

**Comment by superloach** [comment]
I'm currently working on updating to 6.0.0, but for now I wanted to correct the issues I noticed on 4.0.0. It'll likely require similar permissions and dependencies.

**Comment by JL2210** [comment]
Weston 6 is already out? Wow. It's been a long time since I've checked.

**Comment by superloach** [comment]
Yep, released toward the end of March. I'm currently working through a nice little dependency tree for it.

**Comment by JL2210** [comment]
Sigh... Isn't that the best part of package managers?

**Comment by superloach** [comment]
Indeed, it is. Love having to find secret systemd-disabling flags on LFS.

**Comment by JL2210** [comment]
This automates the process of installing Gentoo Prefix on a Chromebook:

https://gist.github.com/JL2210/f51a3d5a7a57900983b706949eba9f74

Dependency trees are no problem. (other than when they don't merge, in which case they're a big problem).

**Comment by superloach** [comment]
Huh, that's neat.

**Comment by JL2210** [comment]
It makes up for `dev_install` and `emerge` being broken on default Chrome OS.

**Comment by superloach** [comment]
Hm, so `weston` depends on `colord`, which depends on `polkit`, which depends on `mozjs-60`. *Fun.*

**Comment by JL2210** [comment]
Are those hard dependencies or just optional? We can leave them out if they're optional.

**Comment by superloach** [comment]
I'd love to avoid some of these, but I'm not sure where to look for Meson flags.

**Comment by JL2210** [comment]
`cat meson.build`.

**Comment by superloach** [comment]
Thanks, I'll give that a go.

**Comment by superloach** [comment]
Alright, so after some fun with Meson options, I've run into a missing header. `linux/input-event-codes.h` seems to be effectively the same as `linux/input.h`, guess I'll patch that?

**Comment by JL2210** [comment]
Yes.

**Comment by JL2210** [comment]
It looks like it was changed right after 3.18 came out, so that should be fine.

**Comment by superloach** [comment]
Alright, I'm pretty sure I've got a working Weston 6.0.0 package, I'll just test for dependencies real quick.

**Comment by superloach** [comment]
Do you want to include freerdp support? That rakes in a lot more dependencies that Weston doesn't normally need.

**Comment by JL2210** [comment]
You can leave it out until somebody complains.

**Comment by superloach** [comment]
Hm, I'm getting this error on most of the Weston demos with 6.0.0.
```
../clients/window.c:5161: window_create_internal: Assertion `custom || display->xdg_shell' failed.
```

**Comment by superloach** [comment]
Tried again on the latest commit, same issues.
